employees/views.py

```diff
@@ -214,7 +214,7 @@ class ReportDetailBase(UpdateView):
 
     def get_initial(self) -> dict:
         initial = super().get_initial()
-        initial.update({"author": self.request.user})
+        initial.update({"author": Report.objects.get(pk=self.kwargs["pk"]).author})
```

Comment:
Report should be already available as `self.object`. If it is not, please get it with `self.get_object()`. This method takes into consideration all updates made ie. in `self.queryset` or `get_queryset` method.

Reply:
Done